app/src/main/java/org/xbmc/kore/ui/widgets/NowPlayingPanel.java

```diff
@@ -0,0 +1,271 @@
+
+        setupButtonClickListeners();
+    }
+
+    public void setOnPanelButtonsClickListener(OnPanelButtonsClickListener listener) {
+        onPanelButtonsClickListener = listener;
+    }
+
+    public void setOnVolumeChangeListener(VolumeLevelIndicator.OnVolumeChangeListener listener) {
+        volumeLevelIndicator.setOnVolumeChangeListener(listener);
+    }
+
+    public void setOnProgressChangeListener(MediaProgressIndicator.OnProgressChangeListener listener) {
+        mediaProgressIndicator.setOnProgressChangeListener(listener);
+    }
+
+    public void setVolume(int volume, boolean muted) {
+        volumeLevelIndicator.setVolume(muted, volume);
+
+        if (muted) {
+            volumeMutedIndicatorButton.setVisibility(View.VISIBLE);
+        } else {
+            volumeMutedIndicatorButton.setVisibility(View.GONE);
+        }
+
+        volumeMutedIndicatorButton.setHighlight(muted);
+        volumeMuteButton.setHighlight(muted);
+    }
+
+    public void setRepeatMode(String repeatMode) {
+        UIUtils.setRepeatButton(repeatModeButton, repeatMode);
+    }
+
+    public void setShuffled(boolean shuffled) {
+        shuffleButton.setHighlight(shuffled);
+    }
+
+    /**
+     * Sets the state of the play button
+     * @param play true if playing, false if paused
+     */
+    public void setPlayButton(boolean play) {
+        UIUtils.setPlayPauseButtonIcon(getContext(), playButton, play);
+    }
+
+    public void setMediaProgress(GlobalType.Time time, GlobalType.Time totalTime) {
+        mediaProgressIndicator.setMaxProgress(totalTime.ToSeconds());
+        mediaProgressIndicator.setProgress(time.ToSeconds());
+    }
+
+    /**
+     * Returns the progression indicator used for media progression
+     * @return
+     */
+    public MediaProgressIndicator getMediaProgress() {
+        return mediaProgressIndicator;
+    }
+
+    /**
+     *
+     * @param speed
+     */
+    public void setSpeed(int speed) {
+        mediaProgressIndicator.setSpeed(speed);
+    }
+
+    public CharSequence getTitle() {
+        return title.getText();
+    }
+
+    public void setTitle(String title) {
+        this.title.setText(title);
+    }
+
+    public void setDetails(String details) {
+        this.details.setText(details);
+    }
+
+    public void setNextPrevVisibility(int visibility) {
+        nextButton.setVisibility(visibility);
+        previousButton.setVisibility(visibility);
+    }
+
+    public void setSquarePoster(boolean square) {
+        if (square) {
+            ViewGroup.LayoutParams layoutParams = poster.getLayoutParams();
+            layoutParams.width = layoutParams.height;
+            poster.setLayoutParams(layoutParams);
+        }
+    }
+
+    public ImageView getPoster() {
+        return poster;
+    }
+
+    private void setupButtonClickListeners() {
+        playButton.setOnClickListener(new OnClickListener() {
```

Comment:
Why not create only one `OnClickListener` and set it as the handle for all these buttons.